tests/test_check_tx.py

```diff
@@ -485,3 +488,308 @@ def test_check_score(difference, expected_result):
         return_value=10**18 + difference,
     ):
         assert check_score(onchain_data, offchain_data) == expected_result
+
+
+@pytest.mark.parametrize(
+    "scenario,expected_result",
+    [
+        # No hooks defined - should pass
+        ("no_hooks", True),
+        # All hooks executed correctly - should pass
+        ("all_hooks_executed", True),
+        # Missing pre-hook - should fail
+        ("missing_pre_hook", False),
+        # Missing post-hook - should fail
+        ("missing_post_hook", False),
+        # Insufficient gas limit - should fail
+        ("insufficient_gas", False),
+        # First fill with pre-hooks - should pass
+        ("first_fill_with_pre_hooks", True),
+        # Subsequent fill without pre-hooks - should pass
+        ("subsequent_fill_no_pre_hooks", True),
+        # Subsequent fill with pre-hooks present (should ideally fail, but passes because
+        # current implementation doesn't validate that pre-hooks are absent on subsequent fills)
+        ("subsequent_fill_with_pre_hooks", True),
+        # Multiple hooks executed correctly - should pass
+        ("multiple_hooks", True),
+    ],
```

Comment:
A suggestion for an extra test case.

```suggestion
        # Multiple orders with the same hook, executed only once - should pass
        ("multiple_orders_same_hook_executed_once", True),
    ],
```